tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -569,22 +567,69 @@ Status StorageManagerCanonical::delete_fragments(
 
   // Write ignore file
   if (commit_uris_to_ignore.size() != 0) {
-    RETURN_NOT_OK(write_commit_ignore_file(array_dir, commit_uris_to_ignore));
+    throw_if_not_ok(write_commit_ignore_file(array_dir, commit_uris_to_ignore));
   }
 
   // Delete fragments and commits
-  st = parallel_for(compute_tp(), 0, fragment_uris.size(), [&](size_t i) {
-    RETURN_NOT_OK(vfs()->remove_dir(fragment_uris[i].uri_));
+  throw_if_not_ok(
+      parallel_for(compute_tp(), 0, fragment_uris.size(), [&](size_t i) {
+        RETURN_NOT_OK(vfs()->remove_dir(fragment_uris[i].uri_));
+        bool is_file = false;
+        RETURN_NOT_OK(vfs()->is_file(commit_uris_to_delete[i], &is_file));
+        if (is_file) {
+          RETURN_NOT_OK(vfs()->remove_file(commit_uris_to_delete[i]));
+        }
+        return Status::Ok();
+      }));
+}
+
+void StorageManagerCanonical::delete_fragments_list(
+    const char* array_name, const std::vector<std::string> fragment_uris) {
+  if (array_name == nullptr) {
+    throw Status_StorageManagerError(
+        "[delete_fragments_list] Array name cannot be null");
```

Comment:
This error shouldn't be `Status`-anything; it doesn't represent anything the user has control over. A check for `nullptr` in the API function should deal with what the user does control. Here we should never see `nullptr`; seeing it is a logic error. I'd recommend throwing `std::invalid_argument` (a subtype of `logic_error`).